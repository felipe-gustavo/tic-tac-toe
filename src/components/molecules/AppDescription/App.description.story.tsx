import { ComponentMeta } from "@storybook/react";

import { AppDescription } from "./AppDescription";

export default {
  title: "Components/AppDescription",
  component: AppDescription,
} as ComponentMeta<typeof AppDescription>;

export const Common = AppDescription.bind({});
