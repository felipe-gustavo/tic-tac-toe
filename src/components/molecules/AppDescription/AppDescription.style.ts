import styled from "styled-components";

export const AppLink = styled.a`
  color: #61dafb;
`;
