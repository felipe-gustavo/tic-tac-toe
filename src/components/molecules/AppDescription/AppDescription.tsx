import { AppLink } from "./AppDescription.style";

export function AppDescription() {
  return (
    <>
      <p>
        Edit <code>src/App.tsx</code> and save to reload.
      </p>
      <AppLink
        href="https://reactjs.org"
        target="_blank"
        rel="noopener noreferrer"
      >
        Learn React
      </AppLink>
    </>
  );
}
