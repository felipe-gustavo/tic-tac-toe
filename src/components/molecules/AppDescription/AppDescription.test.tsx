import { render, screen } from "@testing-library/react";

import { AppDescription } from "./AppDescription";

describe("<AppDescription />", () => {
  it("renders learn react link", () => {
    render(<AppDescription />);
    const linkElement = screen.getByText(/learn react/i);
    expect(linkElement).toBeInTheDocument();
  });

  it("match with screenshot", () => {
    const { baseElement } = render(<AppDescription />);

    expect(baseElement).toMatchSnapshot();
  });
});
