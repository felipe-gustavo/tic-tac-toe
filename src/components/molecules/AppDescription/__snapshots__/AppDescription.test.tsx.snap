// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppDescription /> match with screenshot 1`] = `
<body>
  <div>
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to reload.
    </p>
    <a
      class="sc-bczRLJ jwNrvz"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </div>
</body>
`;
